core/processor/abci.go

```diff
@@ -626,7 +627,13 @@ func (app *App) OnEndBlock(req tmtypes.RequestEndBlock) (ctx context.Context, re
 		logging.String("previous-datetime", vegatime.Format(app.previousTimestamp)),
 	)
 
-	app.epoch.OnBlockEnd(ctx)
+	app.broker.Send(
```

Comment:
I feel like should we not being sending as the last thing we do in`OnEndBlock()` once the engines have finished all their end-block processing and sent out their events?

Reply:
addressed